## Import

In [11]:
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix

## Loading Data (processed)

In [7]:
processed_file_path = '../data/processed/stroke_data_processed.csv'
df = pd.read_csv(processed_file_path)

We have to split our dataset into training and testing dataset.  
(70/30 or 80/20 are good)

In [9]:
# Split the data into features and target
X = df.drop(columns='stroke')
y = df['stroke']

# Split into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [10]:
# Initialize and train the model
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

# Make predictions
y_pred = model.predict(X_test)

# Evaluate the model
print("Classification Report:")
print(classification_report(y_test, y_pred))

print("Confusion Matrix:")
print(confusion_matrix(y_test, y_pred))


Classification Report:
              precision    recall  f1-score   support

           0       0.94      1.00      0.97      1444
           1       0.50      0.01      0.02        89

    accuracy                           0.94      1533
   macro avg       0.72      0.51      0.50      1533
weighted avg       0.92      0.94      0.92      1533

Confusion Matrix:
[[1443    1]
 [  88    1]]


Mmmm.. results are not so good.  
They are actually to predict if a person will not have a stroke (0) with 94% accuracy, but only 50% for stroke (1).